contracts/AirDrop.sol

```diff
@@ -0,0 +1,242 @@
+        );
+        bytes32 structHash = keccak256(
+            abi.encode(DROP_HASH, amount_, deadline_, recepients_)
+        );
+        bytes32 digest = EIP712._hashTypedDataV4(structHash);
+
+        address msgSigner = ECDSA.recover(digest, v_, r_, s_);
+        require(_owner == msgSigner, "invalid signature");
+        require(
+            block.timestamp < deadline_,
+            "Error : signed transaction expired"
+        );
+
+        balanceOfEther[recepients_] += amount_;
+        emit DropEther(recepients_, amount_, deadline_);
+    }
+
+    /**
+     * @dev Change actually ERC20
+     *
+     * @param token_ - address of new ERC20 which connected
+     * to 'this' contract
+     */
+    function updateTokenAddress(address token_) external override onlyOwner {
+        require(
+            token_.code.length > 0,
+            "Error : Incorrect address , only contract address"
+        );
+        token = IERC20(token_);
+        emit NewContractAddress(token_);
+    }
+
+    /**
+     * @dev This function return tokens to 'owner'
+     *
+     * NOTE: withdraw all available tokens
+     */
+    function withdrawTokens() external override onlyOwner {
+        uint256 balance = token.balanceOf(address(this));
+        require(balance > 0, "Error : No availabale tokens to withdraw");
+        emit WithdrawTokens(balance, msg.sender);
+        token.safeTransfer(msg.sender, balance);
+    }
+
+    /**
+     * @dev This function return ether to 'owner'
+     *
+     * NOTE: withdraw all available ether
+     */
+    function withdrawEther() external override onlyOwner {
+        require(
+            address(this).balance > 0,
+            "Error : No availabale ether to withdraw"
+        );
+        emit WithdrawEther(address(this).balance, msg.sender);
+        payable(msg.sender).transfer(address(this).balance);
+    }
+
+    /**
+     * @dev allows to claim available tokens
+     * to the member
+     *
+     * NOTE: withdraw all available tokens , you can call
+     * this function only if you have available tokens
+     */
+    function claimToken() external override {
+        address member = msg.sender;
+        uint256 amountOfTokens = balanceOfTokens[member];
+        require(amountOfTokens > 0, "Error : No available tokens to claim");
+        balanceOfTokens[member] -= amountOfTokens;
+        token.safeTransfer(member, amountOfTokens);
+        emit ClaimToken(amountOfTokens, member);
+    }
+
+    /**
+     * @dev allows to claim available ether
+     * to the member
+     *
+     * NOTE: withdraw all available ether , you can call
+     * this function only if you have available ether
+     */
+    function claimEther() external override {
+        address member = msg.sender;
+        uint256 amountOfEther = balanceOfEther[member];
+        require(amountOfEther > 0, "Error : No available ether to claim");
+        balanceOfEther[member] -= amountOfEther;
+        payable(member).transfer(amountOfEther);
```

Comment:
not use transfer/send to send ether. it's have potential issue in new vesrions. go just use call, or Address library (*) from openzepllin